### PR TITLE
Raise exception when trying to build PyTorch on 32-bit Windows system

### PR DESCRIPTION
**Pull request by malfet**
Makes errors in cases described in #27815 more obvious

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -165,6 +165,8 @@
 import sys
 if sys.version_info < (3,):
     raise Exception("Python 2 has reached end-of-life and is no longer supported by PyTorch.")
+if sys.maxsize.bit_length() == 31:
```
Comment (peterjc123) [inline setup.py:168]: What about those embedded devices?

**Comment by ezyang** [comment]
Is this with reference to https://github.com/pytorch/pytorch/issues/27815 ? I thought the plan on record was just to disable this on Windows...

File: setup.py (hunk 1 above)
Comment (malfet) [reply setup.py:168]: I think all currently supported ARM platforms are actually 64-bit

**Comment by malfet** [comment]
@ezyang I can limit the check to just Windows, but can you think of any other 32-bit platform we might want to support?

**Comment by peterjc123** [comment]
> @ezyang I can limit the check to just Windows, but can you think of any other 32-bit platform we might want to support?

So we are rejecting users to build that by themselves, right?

**Comment by malfet** [comment]
> So we are rejecting users to build that by themselves, right?
Yes, because 32-bit PyTorch build is currently not supported, but if user tried to build it using 32-bit runtime it fails with a confusing error message.

**Review by ezyang — APPROVED** [review]
I might suggest elaborating the error message a bit. The most common reason why bit size is 32 bit is because you have 32 bit Python, but your system might actually be 64 bit, in which case the correct fix is to install 64 bit Python (which is what was the case for the issue people filed.)

**Comment by peterjc123** [comment]
> I might suggest elaborating the error message a bit. The most common reason why bit size is 32 bit is because you have 32 bit Python, but your system might actually be 64 bit, in which case the correct fix is to install 64 bit Python (which is what was the case for the issue people filed.)

I think it will more useful if we patch the v0.1.2 source so that it will present better message if users try to install with 32-bit Python on Windows.

**Pull request closed without merging**